sms_939/models/iap_account.py

```diff
@@ -40,3 +40,7 @@ def _server_env_fields(self):
             }
         )
         return res
+
+    def write(self, vals):
+        # Ensure sudo() is used for elevated permissions
+        return super(IapAccount, self.sudo()).write(vals)
```

Comment:
In that case it's better to add security access to all users instead of doing a sudo here. It's also strange that the SMS accounts needs to be written. Maybe you should debug why the write method is called and maybe fix that instead.

Reply:
i will find out why the write method is being called then thank you.

Reply:
changed the name of the branch to the correct task FIX-T1294-Creating-new-contact